pandas-stubs/core/frame.pyi

```diff
@@ -1660,7 +1660,8 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
         bool_only: _bool | None = ...,
         skipna: _bool = ...,
         **kwargs: Any,
-    ) -> _bool: ...
+    ) -> np.bool: ...
+    # FIXME the type below is not correct, should be pd.Series[np.bool]
```

Comment:
The way I look at it is that we are using `Series[bool]` to correspond to whatever `bool` is stored inside - being a `python` one, a `numpy` one, or even if we use `BooleanDtype`, so I don't think this comment is necessary.

Reply:
Actually the type checker will not accept that, because pd.Series can only be subscribed with S1 which contains bool (the generic boolean from python) but not np.bool.
Yet pd.DataFrame.any will return at runtime pd.Series[np.bool] (but this is not accepted since np.bool is not a subtype of S1).
Let me know if this is not clear,
<img width="1115" alt="image" src="https://github.com/user-attachments/assets/4dceb46d-dd3b-4077-9ec4-8a94bc090b7a" />

Reply:
The type returned from `DataFrame.any()` and `DataFrame.all()` should be `Series[_bool]` .  Then the tests should use that.

Reply:
My issue is that _bool does not contain `np.bool` which is the type we get at runtime. Happy to keep the stubs as is but that would mean that runtime type does not align with static type.
Or we can add np.bool to _bool, open to suggestion.

Reply:
I don't think it matters.  We have checks like this:
```python
check(assert_type(df.any(), "pd.Series[bool]"), pd.Series, np.bool_)
```
So even though `np.bool_` is in the `Series`, we call the type of the `Series[bool]`

It's similar to this:
```python
    check(assert_type(df.value_counts(), "pd.Series[int]"), pd.Series, np.integer)
```
What's inside the series are numpy integers, but we call that `Series[int]`

Reply:
Okay I see your point, let me fix the comments, thanks for the more detailed vision on this issue!